UI streamlining updates to the Settings Page (alert preferences form)

lwu15 opened the pull request:
Quick updates:
- Integrate current preferences into the main preferences box, eliminating the separate box on the left side. Now it's just shown above the checkboxes in just one line for easy reference.
- Update checkboxes accordingly, ensuring they are checked if the current preferences are so.
- Clean up code and add comments for Material UI.
- Add edge cases for email and Slack URL inputs to handle invalid entries, preventing users from entering emails or URLs that are way too long and invalid.

lwu15 committed:
Integrate current preferences for cleaner UI and handle edge cases for email and Slack URL

lwu15 committed:
Merge branch 'dev' into lwu/alerts

hjdjoo commented on src/pages/Settings.jsx:
```diff
@@ -49,32 +52,40 @@ const AlertSettings = () => {
   }, [userInfo?.userID, settingsFeedback]);
-  
 
   const handleSubmit = async (event) => {
```
I noticed a few UX things that could use tweaking during review:

- When a user inputs their password and presses submit, the password field should be cleared.
- If a user has already selected "email" as one of their preferences and wants to add "in-app" as another preference, they are required to re-enter their email.

Can we make it so that saved preferences are independent of the form state? That should make it so that they don't have to re-enter all of their information when they try to update their alert preferences, instead just adding the alerts to their preferences.

lwu15 committed:
Merge branch 'dev' into lwu/alerts

lwu15 committed:
Improve Alert Preferences form functionality and UI

lwu15 commented:
@hjdjoo Thanks for discussing it with me! Add. Changes made:
- Added code to reset the password field after a successful API call for improved security. 
- Conditionally hid the email text field when an email is already saved for better UI.
- Replaced getEmelementByid with useState variables for emailAddress and slackURL for consistency.
- Made minor adjustments to the conditional checks in handleSubmit for improved validation.

lwu15 committed:
Include Beta Alert disclaimer for Email and In-App alert functionalities

lwu15 committed:
Merge branch 'dev' into lwu/alerts

hjdjoo reviewed (approved):
Approving for the time being, but still have issues with adding alert preferences overwriting previous preferences.
